source/compilation/ElabVisitors.h

```diff
@@ -518,4 +518,72 @@ struct DefParamVisitor : public ASTVisitor<DefParamVisitor, false, false> {
     const InstanceSymbol* hierarchyProblem = nullptr;
 };
 
+// This visitor is for adding the only-assigned-on-reset diagnostic to the registers variables that
+// are only assigned on reset
+struct OnlyResetAssignment : public ASTVisitor<OnlyResetAssignment, true, false> {
+    struct AlwaysFFVisitor : public ASTVisitor<AlwaysFFVisitor, true, false> {
+        struct LookupLhsIdentifier : public ASTVisitor<LookupLhsIdentifier, true, true> {
+            explicit LookupLhsIdentifier(string_view name) : name(name) {}
+
+            void handle(const AssignmentExpression& expression) {
+                if (expression.left().kind == ExpressionKind::NamedValue) {
```

Comment:
This check is why this isn't working for writes to struct or array members. You can use the Expression::getSymbolReference() method to get you the symbol referred to by an expression in a more generic way. For example:
```
if (auto symbol = expression.left().getSymbolReference(); symbol && symbol->name == name) { found = true; }
```